guardian/prosemirror-typerighter
# Add an 'ignore matches' predicate

## What does this change?

Adds an 'ignore matches' predicate that the plugin consumer can supply. This predicate receives every match supplied to the plugin on init or as a result of a successful response. If it returns false, the match is not added to the plugin state.

An example use case at the Guardian: in our text editor, our users have the ability to mark text ranges as 'correct'. We don't want to see matches for these ranges.

## How to test

The added tests should give us an idea of how this works.

You could also spin this up in a test environment and add a predicate to check it behaves as expected.

## How can we measure success?

Matches that ... match the predicate are not included in the plugin state.
